### PR TITLE
Support for launching from command line in offline mode

### LINKED ISSUE #1059: Launch Instances Offline From Command Line `-o <name>`
### Role

I play modded Minecraft, with a custom setup that allows me to play local co-op without an internet connection.

### Suggestion

Add in a launch flag for launching instances offline. Using `-l <instance> -o <name> `as a way of launching offline instances with a specific name.

### Benefit

This is so you can do offline LAN games, or quickly spin up offline instances for pure singleplayer.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

This is specifically good for quickly spinning up offline instances. There is a really important use case for this as well.

Using a tiling window manager (or Bismuth on KDE), turning off 'pause on unfocus' in the minecraft config, creating a controller-modded instance for each player, and launching the seperate instances in offline mode, you can have offline splitscreen LAN minecraft.

For everything above, it's scriptable using Bash. Find and replace on text files for configs in both Minecraft and WMs, copying instances and duplicating them, and launching instances... *but not in offline mode*. You have to use the GUI for this, which makes scripting this setup impossible.

I believe that adding the -o flag would be very helpful, especially because if you want to test on a server offline, you can do this:

```
prismlauncher -l Offline1 -o OfflinePlayerOne -s localhost
prismlauncher -l Offline2 -o OfflinePlayerTwo -s localhost
```

### PR DESCRIPTION
**Pull request by michaeltyson**
This allows launching an offline instance with --offline --name=OfflineName. This is useful for playing split screen by creating two side-by-side instances, which is impossible online, unless one is using two separate paid accounts.

With this PR, it makes it possible to launch from a script - otherwise, one has to launch manually, which is a pain, or create offline profiles for each instance, which interferes with some functionality like skins (my autistic son takes great issue with his skin not being visible, when using offline profiles!).

Implementation is based on MultiMC, which supports this feature.

See also #1059  for discussion.

### CODE REVIEW

File: launcher/LaunchController.cpp [hunk 1]
```diff
@@ -187,16 +187,22 @@ void LaunchController::login()
...
-                    QString lastOfflinePlayerName = APPLICATION->settings()->get("LastOfflinePlayerName").toString();
-                    QString usedname = lastOfflinePlayerName.isEmpty() ? m_session->player_name : lastOfflinePlayerName;
-                    QString name = QInputDialog::getText(m_parentWidget, tr("Player name"), message, QLineEdit::Normal, usedname, &ok);
-                    if (!ok) {
-                        tryagain = false;
-                        break;
+                    QString usedname;
+                    if(m_offlineName.isEmpty()) {
+                        QString lastOfflinePlayerName = APPLICATION->settings()->get("LastOfflinePlayerName").toString();
+                        QString usedname = lastOfflinePlayerName.isEmpty() ? m_session->player_name : lastOfflinePlayerName;
```
Comment (Trial97) [inline launcher/LaunchController.cpp]:
```suggestion
                        usedname = lastOfflinePlayerName.isEmpty() ? m_session->player_name : lastOfflinePlayerName;
```

**Review by Trial97 — COMMENTED** [review]
Looks good, but do not forget to format the code.
I use the following command when I want to format the code `find ./launcher -iname "*.h" -o -iname "*.cpp" | xargs clang-format -i`

File: launcher/LaunchController.cpp (hunk 1 above)
Comment (michaeltyson) [reply launcher/LaunchController.cpp]: Great catch! Thank you! Okay, new changes pushed.

**Review by Trial97 — APPROVED** [review]

File: launcher/LaunchController.h [hunk 2]
```diff
@@ -48,14 +48,16 @@ class LaunchController : public Task {
     void executeTask() override;
 
     LaunchController(QObject* parent = nullptr);
-    virtual ~LaunchController(){};
+    virtual ~LaunchController() {};
```
Comment (Trial97) [inline launcher/LaunchController.h]: Revert this change to please the format check:
```suggestion
    virtual ~LaunchController(){};
```

**Comment by Trial97** [comment]
Also a non-blocking suggestion: I would have used only one flag similar to how it is mentioned in the linked issue.

**Comment by michaeltyson** [comment]
> Also a non-blocking suggestion: I would have used only one flag similar to how it is mentioned in the linked issue.

That's a good point - it does seem a tad redundant. I basically followed the MultiMC way of doing it minimise changes for those coming from MultiMC like I have. I feel like there are reasonable arguments on both sides. I honestly can't decide! Maybe I should do it the more succinct way, it's not like it's hard to just read the usage text if you're coming from MultiMC..

**Review by Trial97 — APPROVED** [review]

**Comment by TheKodeToad** [comment]
Please be careful - the reason we do not merge MultiMC changes is not just because of the amount of conflicts there would be, but also incompatible licenses!

**Comment by michaeltyson** [comment]
That’s fair! However, I only used MultiMC as inspiration, and a guide for where to make the changes; the code changes are my own and no copy and pasting was done. I don’t believe there’d be any licensing issues here.

**Review by TheKodeToad — APPROVED** [review]

**Comment by Trial97** [comment]
@michaeltyson  can you rebase this PR(or fix the conflicts)

**Comment by michaeltyson** [comment]
Sure! That should merge now.

**Review by Ryex — APPROVED** [review]
